solidity/contracts/liquidity-protection/LiquidityProtection.sol

```diff
@@ -86,29 +88,34 @@ contract LiquidityProtection is TokenHandler, Utils, Owned, ReentrancyGuard, Tim
      *
      * @param _settings liquidity protection settings
      * @param _store liquidity protection store
+     * @param _store liquidity protection stats
      * @param _networkTokenGovernance network token governance
      * @param _govTokenGovernance governance token governance
      * @param _lastRemoveCheckpointStore last liquidity removal/unprotection checkpoints store
      */
     constructor(
         ILiquidityProtectionSettings _settings,
         ILiquidityProtectionStore _store,
+        ILiquidityProtectionStats _stats,
         ITokenGovernance _networkTokenGovernance,
         ITokenGovernance _govTokenGovernance,
         ICheckpointStore _lastRemoveCheckpointStore
     )
         public
         validAddress(address(_settings))
         validAddress(address(_store))
+        validAddress(address(_stats))
         validAddress(address(_networkTokenGovernance))
         validAddress(address(_govTokenGovernance))
         notThis(address(_settings))
         notThis(address(_store))
+        notThis(address(_stats))
```

Comment:
Really not important (and you weren't aware of it), but you can now use the new `validExternalAddress` modifier which combines `validAddress` and `notThis`

Reply:
@yudilevi : What's the verdict? Use `validExternalAddress`, or we scrap of the `notThis` verification and use `validAddress` (which is what I currently changed the code to, due to a 'stack-too-deep' error)?